examples/object_detection/ios/ObjectDetector/ViewContoller/ViewController.swift

```diff
@@ -0,0 +1,527 @@
+// Copyright 2023 The MediaPipe Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import MediaPipeTasksVision
+import UniformTypeIdentifiers
+import AVKit
+
+class ViewController: UIViewController {
+
+  // MARK: Storyboards Connections
+  @IBOutlet weak var previewView: PreviewView!
+  @IBOutlet weak var overlayView: OverlayView!
+  @IBOutlet weak var addImageButton: UIButton!
+  @IBOutlet weak var cameraUnavailableLabel: UILabel!
+  @IBOutlet weak var imageEmptyLabel: UILabel!
+  @IBOutlet weak var resumeButton: UIButton!
+  @IBOutlet weak var runningModelTabbar: UITabBar!
+  @IBOutlet weak var cameraTabbarItem: UITabBarItem!
+  @IBOutlet weak var photoTabbarItem: UITabBarItem!
+  @IBOutlet weak var bottomSheetViewBottomSpace: NSLayoutConstraint!
+  @IBOutlet weak var bottomViewHeightConstraint: NSLayoutConstraint!
+
+  // MARK: Constants
+  private let inferenceIntervalMs: Double = 300
+  private let inferenceBottomHeight = 220.0
+  private let expandButtonHeight = 41.0
+  private let edgeOffset: CGFloat = 2.0
+  private let labelOffset: CGFloat = 10.0
+  private let displayFont = UIFont.systemFont(ofSize: 14.0, weight: .medium)
+  private let colors = [
+    UIColor.red,
+    UIColor(displayP3Red: 90.0/255.0, green: 200.0/255.0, blue: 250.0/255.0, alpha: 1.0),
+    UIColor.green,
+    UIColor.orange,
+    UIColor.blue,
+    UIColor.purple,
+    UIColor.magenta,
+    UIColor.yellow,
+    UIColor.cyan,
+    UIColor.brown
+  ]
+  private let playerViewController = AVPlayerViewController()
+  private var generator:AVAssetImageGenerator?
+
+  // MARK: Instance Variables
+  private var videoDetectTimer: Timer?
+  private var previousInferenceTimeMs = Date.distantPast.timeIntervalSince1970 * 1000
+  private var maxResults = DefaultConstants.maxResults
+  private var scoreThreshold = DefaultConstants.scoreThreshold
+  private var model = DefaultConstants.model
+  private var runingModel: RunningMode = .liveStream {
+    didSet {
+      objectDetectorHelper = ObjectDetectorHelper(
```

Comment:
@khanhlvg @st-tuanmai 
AFAIK, if you alloc a new objectDetectorHelper every time running mode is set, the previous object will still be in memory since it doesn't go out of scope. You will have to set it to nil explicitly before assigning a new object for ARC to dealloc it properly. Could you memory profile the app and check for unreleased memory?

An alternative is to have 2 separate global object detectors for image and live stream modes. For video mode, each time a video is selected, you can initialize an object detector and make predictions on the frames and deallocate it properly once done. 

The reason why video mode needs initialisation per video is because the detect methods only accept increasing timestamps and if you have one global detector for different videos, this can't be enforced. But in live stream, timestamps are the current time, so having one global detector is okay.

For image mode too, you can initialize an object detector each time a classification is performed, but this is unnecessary IMO.